docs/concepts/architecture.md

```diff
@@ -113,3 +113,43 @@ Nuclio provides the ability to run different versions of the same function simul
 
 The event-source mapping can either specify the exact function version, or utilize a defined alias and thus eliminate the need to change the mapping when a newer version of the function is published.
 
+
+## Processors' internal abstractions
+
+Here is an illustration of the abstraction levels that every event goes through while being processed:
+![event-processing-flow](../../docs/assets/images/internal-processor-abstractions.png)
+
+Thus, every event goes through two EventProcessor levels: the first at the trigger level and the second at the runtime level.
+The EventProcessor exists to generalize the allocation logic and make it abstract.
+
+The event processor allocation logic varies based on the expected behavior. We support two modes:
+Synchronous (default) – Events within a single worker are processed sequentially (FIFO).
+Asynchronous – Events within a single worker are processed concurrently.
+
+Depending on the required behavior, we use different event processor allocators at different levels.
```

Comment:
Depending on the configured mode, Nuclio users different event processor allocators at different levels.